tests/core/test_fileio.py

```diff
@@ -157,18 +157,25 @@ def test_eff_save_restore_from_nemo_file_encrypted(self, asr_model):
         with tempfile.NamedTemporaryFile() as fp:
             filename = fp.name
 
+            # Set key - use checkpoint encryption.
+            NeMoArchive.set_encryption_key("test_key")
+
             # Save model (with random artifact).
             with tempfile.NamedTemporaryFile() as artifact:
                 asr_model.register_artifact(config_path=None, src=artifact.name)
-                asr_model._eff_save_to(save_path=filename, encryption_key="test_key")
+                asr_model.save_to(save_path=filename)
 
-            # Try to read the encrypted file without encryption key.
+            # Try to restore the encrypted archive (weights) without the encryption key.
+            NeMoArchive.set_encryption_key(None)
             with pytest.raises(PermissionError):
                 # Restore the model.
-                asr_model2 = EncDecCTCModel._eff_restore_from(restore_path=filename)
+                asr_model2 = EncDecCTCModel.restore_from(restore_path=filename)
 
             # Restore the model.
-            asr_model3 = EncDecCTCModel._eff_restore_from(restore_path=filename, encryption_key="test_key")
+            NeMoArchive.set_encryption_key("test_key")
```

Comment:
Global setter and getter is not multiprocessing safe, but I guess we don't expect to encrypt decrypt on the fly in a DDP enc anyway so it's ok I think ?

Reply:
Correct. Note this is just setting the encryption key, before any file opening/closing/saving/restoring happens.
The actual "magic" is happening in restore_from, not here.

Reply:
Sounds fine then